### PR TITLE
feat: clarify names in the left nav

### LINKED ISSUE #15: Make the table of content entries more short and concise
Currently it uses `uid` field instead of `name` from each module when inserting entries into the toc. The full `uid` usually is very long and thus makes it very hard for users to see what the entry in the left nav is. Making use of `name` or other similar fields or additional parsing will help.

For bigquery storage, it currently looks like
```
google-cloud-bigquery-storage
|- Overview
|- google.cloud.bigquery_storage_v...
|- google.cloud.bigquery_storage_v...
|- google.cloud.bigquery_storage_v...
|- google.cloud.bigquery_storage_v...
|- google.cloud.bigquery_storage_v...
|- google.cloud.bigquery_storage_v...
|- google.cloud.bigquery_storage_v...
```
before expanding.

### PR DESCRIPTION
**Pull request by dandhlee**
Before you open a pull request, note that this repository is forked from [here](https://github.com/docascode/sphinx-docfx-yaml/).
Unless the issue you're trying to solve is unique to this specific repository, 
please file an issue and/or send changes upstream to the original as well.

__________________________________________________________________

Currently it finds the parent node through `find_node_in_toc_tree` and found the parent by traversing through names and the parent. By using shorter names, it could no longer find the parent, so I had to make use of the additional `uidname` field. This is present in the entire toc structure; making use of only existing `uid` field fails to parse as some entry levels only contain the `name` and `items` field. 

The toc with uid gets used to further determine where entries go, and the one that gets stripped gets put in the `toc.yml` file with the content we want. Overall, we achieve shorter names, for example

```
google-cloud-spanner
|- Overview
|- google.cloud.spanner_v...
|- google.cloud.spanner_v...
|- google.cloud.spanner_v...
|- google.cloud.spanner_v...
|- google.cloud.spanner_v...
|- google.cloud.spanner_v...
|- google.cloud.spanner_v...
|- google.cloud.spanner_v...
```
to
```
google-cloud-spanner
|- Overview
|- database_admin
|- spanner_admin_database_v1.types
|- instance_admin
|- spanner_admin_instance_v1.types
|- batch
|- client
|- database
|- instance
...
```

Fixes #15

### CODE REVIEW

File: docfx_yaml/extension.py [hunk 1]
```diff
@@ -756,6 +764,15 @@ def convert_module_to_package_if_needed(obj):
                     raise ValueError("Unable to dump object\n{0}".format(yaml_data)) from e
 
             file_name_set.add(filename)
+           
+            # Parse the name of the object.
+            # "Types" class will need additional parsing to de-duplicate their names and contain
+            # a portion of their parent name for better disambiguation.
+            node_name = obj['name']
+            if node_name == "types":
+                node_name = ".".join(obj.get('uid').split(".")[-2:])
```
Comment (tbpg) [inline docfx_yaml/extension.py:773]: Are we sure the last two elements always contain what we want?

Comment (dandhlee) [reply docfx_yaml/extension.py:773]: The name is represented in the last portion of the uid, and I just need one more tiebreaker, which the second last item will work out: we can have same module name but different package (`storage_v1.client`, `storage_v1beta2.client`) or different module and same package (`storage_v1.client`, `storage_v1.types`). Any more different than levels of two should appear under different directory structure, and that doesn't need a disambiguation.

However I should fix hardcoding types, and move this algorithm elsewhere so it detects duplicate entries beside `types`.

**Comment by dandhlee** [comment]
`package.module` name can sometimes be long, but hard to shorten it concisely without losing the original context. Will work on that as an enhancement for later on.

**Review by tbpg — APPROVED** [review]
Thank you!

**Pull request merged**